### PR TITLE
Add silver game mode and fix add ingredients only visitor phase

### PR DESCRIPTION
**Pull request by klosyandr**

### CODE REVIEW

File: Cauldron/Assets/Scripts/GameModes/SilverGameMode.cs [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+        }
+
+        private void TryMorningReset(GamePhase phase)
+        {
+            if (phase != GamePhase.Night) return;
+
+            foreach (var temp in ingredients.Dict)
+            {
+                var ingredient = temp.Value;
+                ingredient.gameObject.SetActive(true);
```
Comment (MoreenTissedy) [inline Cauldron/Assets/Scripts/GameModes/SilverGameMode.cs:66]: Grow animation?

File: Cauldron/Assets/Scripts/VisitorManager.cs [hunk 2]
```diff
@@ -91,7 +92,7 @@ public async void Enter(Encounter card)
                     ignoreSavedAttempts = true;
                     break;
             }
-            
+            VisitorEnter?.Invoke();
```
Comment (MoreenTissedy) [inline Cauldron/Assets/Scripts/VisitorManager.cs:95]: Let's rename to VisitorEntering to indicate that the process of entering is not done yet.

File: Cauldron/Assets/Scriptable Objects/GameModes/Silver.asset
Comment (MoreenTissedy) [inline Cauldron/Assets/Scriptable Objects/GameModes/Silver.asset:1]: Maybe some meaningful name, like IngredientUsageMode

File: Cauldron/Assets/Scripts/Cauldron.cs [hunk 3]
```diff
@@ -47,14 +49,15 @@ public void Construct(GameStateMachine gameStateMachine, RecipeProvider recipePr
 
         private void Awake()
         {
-            gameStateMachine.OnChangeState += Clear;
+            gameStateMachine.OnChangeState += ClearAndActivate;
             splash.Stop();
             potionPopup.OnDecline += () => PotionDeclined?.Invoke();
+            potionPopup.OnDecline += () => dropZone.SetActive(true);
```
Comment (MoreenTissedy) [inline Cauldron/Assets/Scripts/Cauldron.cs:55]: Let's make a named method for these two subscriptions and change their places - usually the event is fired at the end of the code or we can get into trouble.

**Review by MoreenTissedy — CHANGES_REQUESTED** [review]

File: Cauldron/Assets/Scripts/VisitorManager.cs (hunk 2 above)
Comment (MoreenTissedy) [reply Cauldron/Assets/Scripts/VisitorManager.cs:95]: Usually it's a bad practice, but okey for us now. 
We should structure events in a way that doesn't disrupt the code flow - mostly in the end of the code, rarely in the beginning. That way we don't care about what happens when the event is triggered.

File: Cauldron/Assets/Scripts/GameModes/SilverGameMode.cs (hunk 1 above)
Comment (klosyandr) [reply Cauldron/Assets/Scripts/GameModes/SilverGameMode.cs:66]: Yes, it works. For ingredients, we have `transform.DOScale(transform.localScale, rotateSpeed).From(Vector3.zero);` in OnEnable().
You are asking about this, aren't you?

**Pull request closed without merging**